src/robotFactory.js

```diff
@@ -1,10 +1,83 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    if (!coords.x) {
+      coords.x = 0;
+      coords.y = 0;
```

Comment:
don't change the coords objects from arguments, create new variables and use them

Reply:
here you assign 0 to object `coords` from function's parameter, that is not okay, create new variable
```suggestion
      const { x = 0, y = 0 } = coords;
```